packages/graphql/src/plugins/nexusMaxExecutionPlugin.ts

```diff
@@ -0,0 +1,74 @@
+import type { DataContext } from '@packages/data-context'
+import { DirectiveLocation, GraphQLDirective } from 'graphql'
+import { plugin } from 'nexus'
+import type { NexusGenFieldTypes } from '../gen/nxs.gen'
+
+interface MaxExecutionConfig {
+  duration: number
+  triggerOnResult?: keyof NexusGenFieldTypes['Subscription']
+}
+
+export const maxExecutionDirective = new GraphQLDirective({
+  name: 'skipMaxExecutionGuard',
+  description: 'Skips any maxExecution guard defined internally on the field',
+  locations: [DirectiveLocation.QUERY],
+  isRepeatable: false,
+})
+
+// Added here because the field is defined automatically by the merged schema
+const CLOUD_EXECUTION: Record<string, Record<string, MaxExecutionConfig>> = {
+  Query: {
+    cloudViewer: {
+      duration: 1000,
+      triggerOnResult: 'cloudViewerChange',
+    },
+  },
+}
+
+export const nexusMaxExecutionPlugin = plugin({
+  name: 'maxExecutionPlugin',
+  description: 'If a maxExecution directive is provided, adds a configured Promise.race to the resolution of the field',
+  fieldDefTypes: `maxExecution?: { duration: number, triggerOnResult?: keyof NexusGenFieldTypes['Subscription'] }`,
+  onCreateFieldResolver (field) {
+    const maxExecutionConfig = (field.fieldConfig.extensions?.nexus?.config?.maxExecution || CLOUD_EXECUTION[field.parentTypeConfig.name]?.[field.fieldConfig.name]) as MaxExecutionConfig | undefined
+
+    if (!maxExecutionConfig) {
+      return
+    }
+
+    return async (source, args, ctx: DataContext, info, next) => {
+      if (info.operation.operation !== 'query' || info.operation.directives?.some((d) => d.name.value === 'skipMaxExecutionGuard')) {
+        return next(source, args, ctx, info)
+      }
+
+      const { duration, triggerOnResult } = maxExecutionConfig
+
+      let timeoutHit = false
+      const RACE_RESULT = {} as const
+
+      const result = await Promise.race([
```

Comment:
I don't think this closes UNIFY-1396 - there is still a 1 second delay when the launchpad initially loads if the user is on a slow network connection. Slow network should not block rendering:

> Desired behavior: All cloud operations happen asynchronously and do not block the launchpad rendering. Tests are added to simulate this condition

I'd expect a solution to this problem to asynchronously "fill in" the missing CloudData fields and never block rendering. Effectively, the `duration` here should always be 0. From what I understand, you have the subscriptions for "filling in" already set up, so would this be a simple change?

Reply:
> Desired behavior: All cloud operations happen asynchronously and do not block the launchpad rendering. Tests are added to simulate this condition

Somewhat, I had it at close to zero and it felt janky, was looking to see if we could strike a balance between time to render and not having it feel jumpy. There's also the difference in that the launchpad is pre-fetching the data for the `cloudViewer` so it's not normally that bad. Maybe 500ms would be a decent middle-ground?

Reply:
I tried adding 

```js
const w = () => {
  return new Promise(res => setTimeout(res, 100000))
}
await w()
```

To simulate a request that never returns (I added to `VersionsDataSource`) but I couldn't even tell I was waiting for 1s. Did you give this a try locally @flotwig? I had the same instinct - thinking 1s is a really huge delay - but I didn't really notice it when I tried it locally.

Reply:
Yeah, the delay was noticeable for me locally. @tgriesser and I met and talked about this, and decided to go ahead and effectively set the `duration` to 0. Two changes will reduce the "jumpiness" that Tim mentioned:

1. Caching the current cloud user (`CachedViewer`) and pre-hydrating it so it's ready at launchpad render time
2. Populate the Git Status information and cache it in-memory so it's ready when the app renders

I think he's going to create a task for (2) and do (1) in this PR, correct me if I'm wrong @tgriesser